### ISSUE TITLE
Better error-handling of crashed queues

### ISSUE THREAD

**Issue by ScuttleSE** [mylar3/mylar3#1399]
**Describe the request**
Today, if a queue crashes due to corrupt data or whatever, it won't restart, and there is no way to easily see if a crash has occurred, and there is no way to remedy a crash, apart from restarting Mylar completely. 

So, the request is basically... better error handling of crashed queues :)

**How did you come up with the idea?**
After leaving Mylar running for a day or two it, without fail, requires a restart to actually do anything due to crashed queues.

**Environment (please complete the following information):**
 - OS: Linux / Docker
 - Branch / commit: master

**Additional information**
Add any other information about the request here.

**Comment by barrelltitor** [comment]
Maybe if we could view the queue somewhere and see what succeeded & errored this would not be an issue. This would be an entirely different request, but just wanted to mention that

**Comment by ScuttleSE** [comment]
The problem is that the whole import-queue crashes silently if it encounters an error. The only way to keep track of it is to constantly monitor the debug-log of the application and restart the whole thing in case of a crash.

And the actual crashes are usually pretty hard to parse. 

```
30-Sep-2023 09:41:21 - ERROR :: mylar.excepthook.315 : Thread-12 (preSearchit) : Uncaught exception: Traceback (most recent call last):
  File "/app/mylar3/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/app/mylar3/mylar/webserve.py", line 6330, in preSearchit
    self.addbyid(sr['comicid'], calledby=True, imported=imported, ogcname=ogcname, nothread=True)
  File "/app/mylar3/mylar/webserve.py", line 1335, in addbyid
    return importer.addComictoDB(comicid, mismatch, None, imported, ogcname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mylar3/mylar/importer.py", line 524, in addComictoDB
    moveit.movefiles(comicid, comlocation, imported)
  File "/app/mylar3/mylar/moveit.py", line 37, in movefiles
    nfilename = renameit['nfilename']
                ~~~~~~~~^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

Traceback (most recent call last):
  File "/app/mylar3/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/app/mylar3/mylar/webserve.py", line 6330, in preSearchit
    self.addbyid(sr['comicid'], calledby=True, imported=imported, ogcname=ogcname, nothread=True)
  File "/app/mylar3/mylar/webserve.py", line 1335, in addbyid
    return importer.addComictoDB(comicid, mismatch, None, imported, ogcname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mylar3/mylar/importer.py", line 524, in addComictoDB
    moveit.movefiles(comicid, comlocation, imported)
  File "/app/mylar3/mylar/moveit.py", line 37, in movefiles
    nfilename = renameit['nfilename']
                ~~~~~~~~^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

```

**Comment by ScuttleSE** [comment]
Almost two years later I'm posting a new debug file here with the same problem

`08-Aug-2025 13:45:19 - INFO    :: mylar.addComictoDB.508 : Thread-23 (preSearchit) : Sucessfully retrieved issue details for Chronicles of Wormwood
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1285 : Thread-23 (preSearchit) : issuedata call references...
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1286 : Thread-23 (preSearchit) : comicid: 18123
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1287 : Thread-23 (preSearchit) : comicname: Chronicles of Wormwood
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1288 : Thread-23 (preSearchit) : comicissues: 6
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1289 : Thread-23 (preSearchit) : calledfrom: None
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1290 : Thread-23 (preSearchit) : issuechk: None
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1291 : Thread-23 (preSearchit) : latestissueinfo: [{'latestiss': '6', 'latestdate': '2007-08-01'}]
08-Aug-2025 13:45:19 - DEBUG   :: mylar.updateissuedata.1292 : Thread-23 (preSearchit) : issuetype: None
08-Aug-2025 13:45:19 - DEBUG   :: mylar.annual_check.1750 : Thread-23 (preSearchit) : checking annual db
08-Aug-2025 13:45:19 - DEBUG   :: mylar.annual_check.1782 : Thread-23 (preSearchit) : [IMPORTER-ANNUAL] - Annual Year:2007
08-Aug-2025 13:45:22 - DEBUG   :: mylar.findComic.155 : Thread-23 (preSearchit) : there are 0 search results...
08-Aug-2025 13:45:22 - DEBUG   :: mylar.updateissuedata.1327 : Thread-23 (preSearchit) : Finished Annual checking.
08-Aug-2025 13:45:22 - INFO    :: mylar.updateissuedata.1345 : Thread-23 (preSearchit) : Now adding/updating issues for Chronicles of Wormwood
08-Aug-2025 13:45:22 - DEBUG   :: mylar.updateissuedata.1604 : Thread-23 (preSearchit) : initiating issue updating - info & status
08-Aug-2025 13:45:22 - INFO    :: mylar.issue_collection.1070 : Thread-23 (preSearchit) : nostatus: False
08-Aug-2025 13:45:22 - INFO    :: mylar.issue_collection.1071 : Thread-23 (preSearchit) : issuedata: [{'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '106176', 'ComicName': 'Chronicles of Wormwood', 'IssueName': None, 'Issue_Number': '1', 'IssueDate': '2007-01-24', 'ReleaseDate': '2007-01-31', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 1000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/88552-18123-106176-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/88552-18123-106176-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '106973', 'ComicName': 'Chronicles of Wormwood', 'IssueName': 'Holy Orders', 'Issue_Number': '2', 'IssueDate': '2007-02-01', 'ReleaseDate': '2007-03-14', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 2000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/98978-18123-106973-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/98978-18123-106973-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '109469', 'ComicName': 'Chronicles of Wormwood', 'IssueName': "Knocking on Heaven's Door", 'Issue_Number': '3', 'IssueDate': '2007-04-01', 'ReleaseDate': '2007-05-09', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 3000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/120027-18123-109469-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/120027-18123-109469-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '110772', 'ComicName': 'Chronicles of Wormwood', 'IssueName': "Like A Bat Out Of Hell I'll Be Gone", 'Issue_Number': '4', 'IssueDate': '2007-06-20', 'ReleaseDate': '2007-06-20', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 4000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/162576-18123-110772-3-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/162576-18123-110772-3-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '111350', 'ComicName': 'Chronicles of Wormwood', 'IssueName': 'The Centre Cannot Hold', 'Issue_Number': '5', 'IssueDate': '2007-07-05', 'ReleaseDate': '2007-07-05', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 5000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/147063-18123-111350-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/147063-18123-111350-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '112284', 'ComicName': 'Chronicles of Wormwood', 'IssueName': 'For The Former Things Are Passed Away', 'Issue_Number': '6', 'IssueDate': '2007-07-01', 'ReleaseDate': '2007-08-01', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 6000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/164393-18123-112284-4-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/164393-18123-112284-4-chronicles-of-wormwo.jpg'}]
08-Aug-2025 13:45:22 - DEBUG   :: mylar.update_metadata.66 : Thread-23 (preSearchit) : metainfo_loaded: {'version': '1.0.2', 'metadata': {'type': 'comicSeries', 'publisher': 'Avatar Press', 'imprint': None, 'name': 'Chronicles of Wormwood', 'comicid': 18123, 'year': 2007, 'description_text': "Meet Danny Wormwood. He's a successful television executive living in New-York, he's got a live-in girlfriend, some internet hobbies, and a pet rabbit who talks. And he's the son of Satan, the Anti-Christ…But he doesn't want to follow blindly in his father's footsteps, or do what some old prophecy dictates. He wants to take is own path, no matter how uncertain it is.", 'description_formatted': "Meet Danny Wormwood. He's a successful television executive living in New-York, he's got a live-in girlfriend, some internet hobbies, and a pet rabbit who talks. And he's the son of Satan, the Anti-Christ…But he doesn't want to follow blindly in his father's footsteps, or do what some old prophecy dictates. He wants to take is own path, no matter how uncertain it is.", 'volume': None, 'booktype': 'Print', 'age_rating': None, 'collects': None, 'comic_image': 'https://comicvine.gamespot.com/a/uploads/scale_large/0/787/88552-18123-106176-1-chronicles-of-wormwo.jpg', 'total_issues': 6, 'publication_run': 'January 2007 - August 2007', 'status': 'Ended'}}
08-Aug-2025 13:45:22 - DEBUG   :: mylar.update_metadata.191 : Thread-23 (preSearchit) : Successfully written series.json file to /comics/Library/Avatar Press/Chronicles of Wormwood (2007)
08-Aug-2025 13:45:22 - INFO    :: mylar.issue_collection.1070 : Thread-23 (preSearchit) : nostatus: False
08-Aug-2025 13:45:22 - INFO    :: mylar.issue_collection.1071 : Thread-23 (preSearchit) : issuedata: [{'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '106176', 'ComicName': 'Chronicles of Wormwood', 'IssueName': None, 'Issue_Number': '1', 'IssueDate': '2007-01-24', 'ReleaseDate': '2007-01-31', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 1000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/88552-18123-106176-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/88552-18123-106176-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '106973', 'ComicName': 'Chronicles of Wormwood', 'IssueName': 'Holy Orders', 'Issue_Number': '2', 'IssueDate': '2007-02-01', 'ReleaseDate': '2007-03-14', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 2000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/98978-18123-106973-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/98978-18123-106973-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '109469', 'ComicName': 'Chronicles of Wormwood', 'IssueName': "Knocking on Heaven's Door", 'Issue_Number': '3', 'IssueDate': '2007-04-01', 'ReleaseDate': '2007-05-09', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 3000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/120027-18123-109469-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/120027-18123-109469-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '110772', 'ComicName': 'Chronicles of Wormwood', 'IssueName': "Like A Bat Out Of Hell I'll Be Gone", 'Issue_Number': '4', 'IssueDate': '2007-06-20', 'ReleaseDate': '2007-06-20', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 4000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/162576-18123-110772-3-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/162576-18123-110772-3-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '111350', 'ComicName': 'Chronicles of Wormwood', 'IssueName': 'The Centre Cannot Hold', 'Issue_Number': '5', 'IssueDate': '2007-07-05', 'ReleaseDate': '2007-07-05', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 5000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/147063-18123-111350-1-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/147063-18123-111350-1-chronicles-of-wormwo.jpg'}, {'ComicID': '18123', 'SeriesStatus': 'Active', 'IssueID': '112284', 'ComicName': 'Chronicles of Wormwood', 'IssueName': 'For The Former Things Are Passed Away', 'Issue_Number': '6', 'IssueDate': '2007-07-01', 'ReleaseDate': '2007-08-01', 'DigitalDate': '0000-00-00', 'Int_IssueNumber': 6000, 'AltIssueNumber': None, 'ImageURL': 'https://comicvine.gamespot.com/a/uploads/scale_small/0/787/164393-18123-112284-4-chronicles-of-wormwo.jpg', 'ImageURL_ALT': 'https://comicvine.gamespot.com/a/uploads/scale_medium/0/787/164393-18123-112284-4-chronicles-of-wormwo.jpg'}]
08-Aug-2025 13:45:22 - DEBUG   :: mylar.addComictoDB.547 : Thread-23 (preSearchit) : no update required - lastissueid [112284] = latestissueid [112284]
08-Aug-2025 13:45:22 - INFO    :: mylar.addComictoDB.572 : Thread-23 (preSearchit) : Updating complete for: Chronicles of Wormwood
08-Aug-2025 13:45:22 - INFO    :: mylar.addComictoDB.585 : Thread-23 (preSearchit) : Mass import - Move files
08-Aug-2025 13:45:22 - DEBUG   :: mylar.movefiles.18 : Thread-23 (preSearchit) : comlocation is : /comics/Library/Avatar Press/Chronicles of Wormwood (2007)
08-Aug-2025 13:45:22 - DEBUG   :: mylar.movefiles.19 : Thread-23 (preSearchit) : original comicname is : Chronicles of Wormwood
08-Aug-2025 13:45:22 - DEBUG   :: mylar.movefiles.35 : Thread-23 (preSearchit) : Renaming files according to configuration details : $Series $Annual #$Issue ($Year)
08-Aug-2025 13:45:22 - DEBUG   :: mylar.rename_param.290 : Thread-23 (preSearchit) : <class 'str'>
08-Aug-2025 13:45:22 - DEBUG   :: mylar.rename_param.291 : Thread-23 (preSearchit) : <class 'NoneType'>
08-Aug-2025 13:45:22 - DEBUG   :: mylar.rename_param.292 : Thread-23 (preSearchit) : comicid: 18123
08-Aug-2025 13:45:22 - DEBUG   :: mylar.rename_param.293 : Thread-23 (preSearchit) : issue# as per cv: 0
08-Aug-2025 13:45:22 - DEBUG   :: mylar.rename_param.315 : Thread-23 (preSearchit) : annualize is None
08-Aug-2025 13:45:22 - ERROR   :: mylar.rename_param.334 : Thread-23 (preSearchit) : Invalid Issue_Number - please validate.
08-Aug-2025 13:45:22 - ERROR   :: mylar.excepthook.315 : Thread-23 (preSearchit) : Uncaught exception: Traceback (most recent call last):
  File "/app/mylar3/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/app/mylar3/mylar/webserve.py", line 6182, in preSearchit
    self.addbyid(comicinfo['ComicID'], calledby=True, imported=imported, ogcname=comicinfo['ComicName'], nothread=True)
  File "/app/mylar3/mylar/webserve.py", line 1494, in addbyid
    return importer.addComictoDB(comicid, mismatch, None, imported, ogcname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mylar3/mylar/importer.py", line 586, in addComictoDB
    moveit.movefiles(comicid, comlocation, imported)
  File "/app/mylar3/mylar/moveit.py", line 37, in movefiles
    nfilename = renameit['nfilename']
                ~~~~~~~~^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

`

**Comment by ScuttleSE** [comment]
Here is another crashed import session

[carepackage_v0.8.2(1).zip](https://github.com/user-attachments/files/21688078/carepackage_v0.8.2.1.zip)

**Comment by barrelltitor** [comment]
This should be a simple fix though it will fail the item

This is because you have no issue # but the code expects an issue #. I can make it if it has no issue number it will just not move it and leave it there  since this also means it is most likely not a proper match

**Comment by ScuttleSE** [comment]
Instead of focusing on the particular problem with this book, why not have a logic where any troublesome book that Mylar can't handle, for any reason, be moved to a separate folder, away from the books that are to be processed, instead of silently crashing?

**Comment by barrelltitor** [comment]
If a crash happens, that is what needs fixed, not a wrapper that just ignores the problem and restarts

The logic is already there to gracefully pass books it cannot handle, the problem here is a bug in the code, which seems to be the same bug in all of your carepackages so fixing it would indeed fix this issue

Feel free to report away all the individual bugs you find, and I promise you I will fix them 

If you have one of these files that causes the crash please send it to me on Discord so I can have a look and play around with it to see what's going on. I suspect there's a problem in the metadata, but I need to see what it is to fix it
